environment/src/main/scala/ru/primetalk/funtik/environment/geom2d/vectors.scala

```diff
@@ -0,0 +1,95 @@
+package ru.primetalk.funtik.environment.geom2d
+
+import spire.algebra.{CModule, CRing, Field, VectorSpace}
+import spire.sp
+
+sealed trait Vector[Axis, Size <: Int]
+
+case class Vector2d[@sp(Int, Double) Axis](x: Axis, y: Axis) extends Vector[Axis, 2] {
+  def _1: Axis = x
+  def _2: Axis = y
+  def transpose: Vector2d[Axis] =
+    Vector2d(y, x)
+}
+
+case class Vector3d[@sp(Int, Double) Axis](x: Axis, y: Axis, z: Axis) extends Vector[Axis, 3] {
+  def _1: Axis = x
```

Comment:
Do we need this aliases?

Reply:
We may remove them as soon as we remove them in other places.